tests/platform_tests/test_link_down.py

```diff
@@ -82,41 +74,45 @@ def fanout_hosts_and_ports(fanouthosts, duts_and_ports):
             logger.info("Interface {} on fanout {} (os type {}) map to interface {} on duthost {}"
                         .format(fanout_port, fanout.hostname, fanout.get_fanout_os(), port, duthost.hostname))
             if fanout in fanout_and_ports.keys():
-                fanout_and_ports[fanout].append(fanout_port)
+                fanout_and_ports[fanout].add(fanout_port)
             else:
-                fanout_and_ports[fanout] = [fanout_port]
+                fanout_and_ports[fanout] = {fanout_port}
     return fanout_and_ports
     
 
-def is_link_down(fanout, port):
+def is_link_down(fanout, ports):
     """
-        Either oper/admin status is down meaning link is down
+        Input:
+            ports: set of ports on this fanout
+        Returns:
+            True: if all ports are down
+            False: if any port is up
     """
-    logger.info("Checking interface {} down status on fanout host {}".format(port, fanout.hostname))
-    return fanout.is_intf_status_down(port)
+    return fanout.is_intf_status_down(ports)
 
 
-def is_link_up(fanout, port):
-    logger.info("Checking interface {} up status on fanout host {}".format(port, fanout.hostname))
-    return not fanout.is_intf_status_down(port)
+def is_link_up(fanout, ports):
+    """
+        Returns:
+            True: if all ports are up
+            False: if any port is down
+    """
+    return not fanout.is_intf_status_down(ports)
```

Comment:
The logic in is_intf_status_down() is such that if one of the interfaces is "up" it will return "False".
Let us take a scenario where first interface is "up" but rest are "down", ["up", "down", "down"].
is_intf_status_down() will return "False" in this case, and is_link_up() will return "True" but actually not all interfaces are "up".
is_intf_status_down() could be modified as check_intf_status(ports, status) and status could be up or down.

Reply:
thanks @SuvarnaMeenakshi for reviewing and catching this!
These 2 functions were used to check per port, now it's checking all ports and logic has changed.
I think it would be cleaner to change to all_links_up(ports) and all_links_down(ports), and call separate functions based on OS type, instead of using the `status` flag?

Reply:
Which ever you think will be clean and help to re-use in future, you could go with that.

Reply:
done, thanks!